### PR TITLE
Completely remove old AffectedCommit code.

### PR DESCRIPTION
**Pull request by oliverchang**
Part of #951.

### CODE REVIEW

File: docker/worker/worker.py [hunk 1]
```diff
@@ -520,12 +515,7 @@ def _do_update(self, source_repo, repo, vulnerability, relative_path,
 
     bug.put()
 
-    # Temporary migration exclusion: Don't write legacy Linux AffectedComit
-    # entries as they cause Datastore scaling issues.
-    write_legacy = not _is_linux_ecosystem(vulnerability)
```
Comment (michaelkedar) [inline docker/worker/worker.py:525]: Should `_is_linux_ecosystem()` also be removed?

File: docker/worker/worker_test.py [hunk 2]
```diff
@@ -145,15 +145,6 @@ def setUp(self):
...
 
-    should_be_deleted = osv.AffectedCommit(
-        id='OSV-2020-1337-abcd',
-        bug_id='OSV-2020-1337',
-        commit='abcd',
-        project='project',
-        ecosystem='ecosystem',
-        public=False)
-    should_be_deleted.put()
-
```
Comment (michaelkedar) [inline docker/worker/worker_test.py:156]: Is there no equivalent new version of this?
(I don't really know what this was testing...)

**Review by michaelkedar — APPROVED** [review]
LGTM

File: docker/worker/worker_test.py (hunk 2 above)
Comment (oliverchang) [reply docker/worker/worker_test.py]: Good catch, this was just checking if the new analysis overwrites previous results. I'll add the new equivalent here and a comment to explain.

**Pull request merged**